core/src/com/unciv/models/ruleset/unique/UniqueTriggerActivation.kt

```diff
@@ -116,8 +117,32 @@ object UniqueTriggerActivation {
                 val event = ruleset.events[unique.params[0]] ?: return null
                 val choices = event.getMatchingChoices(stateForConditionals)
                     ?: return null
+
+                fun calculateEventChoiceWeight(choice: EventChoice): Float {
+                    var weight = 1f
+                    for (modifier in choice.modifierObjects) {
+                        if (modifier.type == UniqueType.AIPriorityModifier && modifier.conditionalsApply(stateForConditionals)) {
+                            weight += unique.params[0].toFloat() / 100f
```

Comment:
We'll probably want to remove the `/100f` here since we aren't working with percents any more.